website/docs/guides/best-practices/how-we-structure/3-intermediate.md

```diff
@@ -0,0 +1,99 @@
+:::
+
+### Intermediate: Models
+
+Below is the lone intermediate model from our small example project. This represents an excellent use case per our principles above, serving a clear single purpose: grouping and pivoting a staging model to different grain. It utilizes a bit of Jinja to make the model DRY-er (striving to be DRY applies to the code we write inside a single model in addition to transformations across the codebase), but don’t be intimidated if you’re not quite comfortable with Jinja yet. Looking at the name of the CTE, `pivot_and_aggregate_payments_to_order_grain` we get a very clear idea of what’s happening inside this block. By descriptively labeling the transformations happening inside our CTEs within model, just as we do with our files and folders, even a stakeholder who doesn’t know SQL would be able to grasp the purpose of this section, if not the code. As you begin to write more complex transformations moving out of the staging layer, keep this idea in mind. In the same way our models connect into a DAG and tell the story of our transformations on a macro scale, CTEs can do this on a smaller scale inside our model files.
+
+```sql
+-- int_payments_pivoted_to_orders.sql
+
+{%- set payment_methods = ['bank_transfer','credit_card','coupon','gift_card'] -%}
+ 
+with 
+
+payments as (
+
+   select * from {{ ref('stg_stripe__payments') }}
+
+),
+ 
+pivot_and_aggregate_payments_to_order_grain as (
+   
+   select
+      order_id,     
+      {% for payment_method in payment_methods -%}
+ 
+         sum(
+            case
+               when payment_method = '{{ payment_method }}' and
+                    status = 'success' 
+               then amount 
+               else 0 
+            end
+         ) as {{ payment_method }}_amount,
+
+      {%- endfor %}
+      sum(case when status = 'success' then amount end) as total_amount
+
+   from payments
+
+   group by 1
+
+)
+ 
+select * from pivot_and_aggregate_payments_to_order_grain
+```
+
+- ❌ **Exposed to end users.** Intermediate models should generally not be exposed in the main production schema. They are not intended for output to final targets like dashboards or applications, so it’s best to keep them separated from models that are so you can more easily control data governance and discoverability.
+- ✅ **Materialized ephemerally.** Considering the above, one popular option is to default to intermediate models being materialized ephemerally. This is generally the best place to start for simplicity. It will keep unnecessary models out of your warehouse with minimum configuration. Keep in mind though that the simplicity of ephemerals does translate a bit more difficulty in troubleshooting, as they’re interpolated into the models that `ref` them, rather than existing on their own in a way that you can view the output of.
```

Comment:
```suggestion
- ✅ **Materialized ephemerally.** Considering the above, one popular option is to default to intermediate models being materialized [ephemerally](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/materializations#ephemeral). This is generally the best place to start for simplicity. It will keep unnecessary models out of your warehouse with minimum configuration. Keep in mind though that the simplicity of ephemerals does translate a bit more difficulty in troubleshooting, as they’re interpolated into the models that `ref` them, rather than existing on their own in a way that you can view the output of.
```